meson_cpu/x86/meson.build

```diff
@@ -60,7 +60,7 @@ FMA3 = mod_features.new(
   test_code: files(source_root + '/numpy/distutils/checks/cpu_fma3.c')[0]
 )
 AVX2 = mod_features.new(
-  'AVX2', 25, implies: F16C, args: '-mavx2',
+  'AVX2', 25, implies: F16C, args: '-march=skylake',
```

Comment:
I wonder if this is more related to implying `-mtune=skylake`? Highway relies on the compiler to do some optimisations, and I do not know what `-mtune=generic` does with just `-mavx2` 🤔

Reply:
Just looked at this a bit more deeply, and it looks like with just `-mavx2` you don't get `HWY_AVX2`, these flags worked to get `HWY_AVX2`:

```
-mpclmul -maes -mavx -mavx2 -mbmi -mbmi2 -mfma -mf16c
```

I assume they're all implied by `haswell` but `avx2` is actually way more limiting. Is there a processor which supports `avx2` without all these things? 🤔

Reply:
yet, the surprising this was using `-march=haswell` makes the performance a lot worse (nearly 4x slower). I need some input from @jan-wassenberg to see what is happening here.

Reply:
You are exactly right, we require more flags than just -march=haswell: also -maes, then it is sufficient.
Unfortunately there are some very few Haswells without AES, so we do check for that.

Reply:
Out of curiosity: why do you need `-maes` for HWY_AVX2 if you aren't using any AES related instructions?

Reply:
AESRound is a supported Highway op; we do not know whether users will use it. I suppose the options are to detect at runtime (but that would add considerable overhead), or add another target for AVX2 \ AES, but this is very rare, so not worthwhile, right?